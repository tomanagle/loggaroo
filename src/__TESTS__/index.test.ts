import logger from '../index';

describe('Logger test', () => {
  it('Should call the info method', () => {
    expect.assertions(1);
    const message = logger.info('Test info message');
    expect(message).toMatchSnapshot();
  });

  it('Should call the error method', () => {
    expect.assertions(1);
    const message = logger.error('Test error message');
    expect(message).toMatchSnapshot();
  });
});
