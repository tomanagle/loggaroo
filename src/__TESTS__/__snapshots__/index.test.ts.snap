// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logger test Should call the error method 1`] = `
DerivedLogger {
  "_events": Object {
    "data": Array [
      [Function],
      [Function],
      [Function],
    ],
    "end": Array [
      [Function],
      [Function],
      [Function],
    ],
    "prefinish": [Function],
  },
  "_eventsCount": 3,
  "_maxListeners": undefined,
  "_readableState": ReadableState {
    "awaitDrain": 0,
    "buffer": BufferList {
      "head": null,
      "length": 0,
      "tail": null,
    },
    "decoder": null,
    "defaultEncoding": "utf8",
    "destroyed": false,
    "emitClose": true,
    "emittedReadable": false,
    "encoding": null,
    "endEmitted": false,
    "ended": false,
    "flowing": true,
    "highWaterMark": 16,
    "length": 0,
    "needReadable": true,
    "objectMode": true,
    "pipes": Array [
      Console {
        "__winstonerror": [Function],
        "__winstonwarn": [Function],
        "_events": Object {
          "close": [Function],
          "drain": [Function],
          "error": Array [
            [Function],
            [Function],
          ],
          "finish": [Function],
          "unpipe": Array [
            [Function],
            [Function],
          ],
          "warn": [Function],
        },
        "_eventsCount": 6,
        "_maxListeners": 30,
        "_writableState": WritableState {
          "bufferProcessing": false,
          "bufferedRequest": null,
          "bufferedRequestCount": 0,
          "corked": 0,
          "corkedRequestsFree": CorkedRequest {
            "entry": null,
            "finish": [Function],
            "next": null,
          },
          "decodeStrings": true,
          "defaultEncoding": "utf8",
          "destroyed": false,
          "ended": false,
          "ending": false,
          "errorEmitted": false,
          "finalCalled": false,
          "finished": false,
          "highWaterMark": 16,
          "lastBufferedRequest": null,
          "length": 0,
          "needDrain": false,
          "objectMode": true,
          "onwrite": [Function],
          "pendingcb": 2,
          "prefinished": false,
          "sync": false,
          "writecb": null,
          "writelen": 0,
          "writing": false,
        },
        "consoleWarnLevels": Object {},
        "domain": null,
        "eol": "
",
        "format": Format {
          "Format": [Function],
          "options": Object {},
        },
        "handleExceptions": undefined,
        "level": undefined,
        "levels": Object {
          "debug": 5,
          "error": 0,
          "http": 3,
          "info": 2,
          "silly": 6,
          "verbose": 4,
          "warn": 1,
        },
        "name": "console",
        "parent": [Circular],
        "silent": undefined,
        "stderrLevels": Object {},
        "writable": true,
      },
      File {
        "__winstonerror": [Function],
        "__winstonwarn": [Function],
        "_basename": "info.log",
        "_created": 0,
        "_drain": false,
        "_ending": false,
        "_events": Object {
          "close": [Function],
          "drain": [Function],
          "error": Array [
            [Function],
            [Function],
          ],
          "finish": [Function],
          "unpipe": Array [
            [Function],
            [Function],
          ],
          "warn": [Function],
        },
        "_eventsCount": 6,
        "_maxListeners": undefined,
        "_onError": [Function],
        "_opening": true,
        "_pendingSize": 94,
        "_size": 0,
        "_stream": PassThrough {
          "_events": Object {
            "prefinish": [Function],
          },
          "_eventsCount": 1,
          "_maxListeners": 30,
          "_readableState": ReadableState {
            "awaitDrain": 0,
            "buffer": BufferList {
              "head": Object {
                "data": Object {
                  "data": Array [
                    49,
                    50,
                    45,
                    70,
                    101,
                    98,
                    45,
                    50,
                    48,
                    49,
                    57,
                    32,
                    48,
                    56,
                    58,
                    53,
                    55,
                    58,
                    52,
                    56,
                    32,
                    105,
                    110,
                    102,
                    111,
                    58,
                    32,
                    84,
                    101,
                    115,
                    116,
                    32,
                    105,
                    110,
                    102,
                    111,
                    32,
                    109,
                    101,
                    115,
                    115,
                    97,
                    103,
                    101,
                    13,
                    10,
                  ],
                  "type": "Buffer",
                },
                "next": Object {
                  "data": Object {
                    "data": Array [
                      49,
                      50,
                      45,
                      70,
                      101,
                      98,
                      45,
                      50,
                      48,
                      49,
                      57,
                      32,
                      48,
                      56,
                      58,
                      53,
                      55,
                      58,
                      52,
                      56,
                      32,
                      101,
                      114,
                      114,
                      111,
                      114,
                      58,
                      32,
                      84,
                      101,
                      115,
                      116,
                      32,
                      101,
                      114,
                      114,
                      111,
                      114,
                      32,
                      109,
                      101,
                      115,
                      115,
                      97,
                      103,
                      101,
                      13,
                      10,
                    ],
                    "type": "Buffer",
                  },
                  "next": null,
                },
              },
              "length": 2,
              "tail": Object {
                "data": Object {
                  "data": Array [
                    49,
                    50,
                    45,
                    70,
                    101,
                    98,
                    45,
                    50,
                    48,
                    49,
                    57,
                    32,
                    48,
                    56,
                    58,
                    53,
                    55,
                    58,
                    52,
                    56,
                    32,
                    101,
                    114,
                    114,
                    111,
                    114,
                    58,
                    32,
                    84,
                    101,
                    115,
                    116,
                    32,
                    101,
                    114,
                    114,
                    111,
                    114,
                    32,
                    109,
                    101,
                    115,
                    115,
                    97,
                    103,
                    101,
                    13,
                    10,
                  ],
                  "type": "Buffer",
                },
                "next": null,
              },
            },
            "decoder": null,
            "defaultEncoding": "utf8",
            "destroyed": false,
            "emitClose": true,
            "emittedReadable": true,
            "encoding": null,
            "endEmitted": false,
            "ended": false,
            "flowing": null,
            "highWaterMark": 16384,
            "length": 94,
            "needReadable": false,
            "objectMode": false,
            "pipes": null,
            "pipesCount": 0,
            "readableListening": false,
            "reading": false,
            "readingMore": true,
            "resumeScheduled": false,
            "sync": false,
          },
          "_transformState": Object {
            "afterTransform": [Function],
            "needTransform": true,
            "transforming": false,
            "writecb": null,
            "writechunk": null,
            "writeencoding": "buffer",
          },
          "_writableState": WritableState {
            "bufferProcessing": false,
            "bufferedRequest": null,
            "bufferedRequestCount": 0,
            "corked": 0,
            "corkedRequestsFree": CorkedRequest {
              "entry": null,
              "finish": [Function],
              "next": null,
            },
            "decodeStrings": true,
            "defaultEncoding": "utf8",
            "destroyed": false,
            "emitClose": true,
            "ended": false,
            "ending": false,
            "errorEmitted": false,
            "finalCalled": false,
            "finished": false,
            "highWaterMark": 16384,
            "lastBufferedRequest": null,
            "length": 0,
            "needDrain": false,
            "objectMode": false,
            "onwrite": [Function],
            "pendingcb": 2,
            "prefinished": false,
            "sync": false,
            "writecb": null,
            "writelen": 0,
            "writing": false,
          },
          "allowHalfOpen": true,
          "domain": null,
          "readable": true,
          "writable": true,
        },
        "_writableState": WritableState {
          "bufferProcessing": false,
          "bufferedRequest": null,
          "bufferedRequestCount": 0,
          "corked": 0,
          "corkedRequestsFree": CorkedRequest {
            "entry": null,
            "finish": [Function],
            "next": null,
          },
          "decodeStrings": true,
          "defaultEncoding": "utf8",
          "destroyed": false,
          "ended": false,
          "ending": false,
          "errorEmitted": false,
          "finalCalled": false,
          "finished": false,
          "highWaterMark": 16,
          "lastBufferedRequest": null,
          "length": 0,
          "needDrain": false,
          "objectMode": true,
          "onwrite": [Function],
          "pendingcb": 2,
          "prefinished": false,
          "sync": false,
          "writecb": null,
          "writelen": 0,
          "writing": false,
        },
        "dirname": ".",
        "domain": null,
        "eol": "
",
        "filename": "info.log",
        "format": Format {
          "Format": [Function],
          "options": Object {},
        },
        "handleExceptions": undefined,
        "level": "info",
        "levels": Object {
          "debug": 5,
          "error": 0,
          "http": 3,
          "info": 2,
          "silly": 6,
          "verbose": 4,
          "warn": 1,
        },
        "maxFiles": null,
        "maxsize": null,
        "name": "file",
        "options": Object {
          "flags": "a",
        },
        "parent": [Circular],
        "rotationFormat": false,
        "silent": undefined,
        "tailable": false,
        "writable": true,
        "zippedArchive": false,
      },
      File {
        "__winstonerror": [Function],
        "__winstonwarn": [Function],
        "_basename": "errors.log",
        "_created": 0,
        "_drain": false,
        "_ending": false,
        "_events": Object {
          "close": [Function],
          "drain": [Function],
          "error": Array [
            [Function],
            [Function],
          ],
          "finish": [Function],
          "unpipe": Array [
            [Function],
            [Function],
          ],
          "warn": [Function],
        },
        "_eventsCount": 6,
        "_maxListeners": undefined,
        "_onError": [Function],
        "_opening": true,
        "_pendingSize": 48,
        "_size": 0,
        "_stream": PassThrough {
          "_events": Object {
            "prefinish": [Function],
          },
          "_eventsCount": 1,
          "_maxListeners": 30,
          "_readableState": ReadableState {
            "awaitDrain": 0,
            "buffer": BufferList {
              "head": Object {
                "data": Object {
                  "data": Array [
                    49,
                    50,
                    45,
                    70,
                    101,
                    98,
                    45,
                    50,
                    48,
                    49,
                    57,
                    32,
                    48,
                    56,
                    58,
                    53,
                    55,
                    58,
                    52,
                    56,
                    32,
                    101,
                    114,
                    114,
                    111,
                    114,
                    58,
                    32,
                    84,
                    101,
                    115,
                    116,
                    32,
                    101,
                    114,
                    114,
                    111,
                    114,
                    32,
                    109,
                    101,
                    115,
                    115,
                    97,
                    103,
                    101,
                    13,
                    10,
                  ],
                  "type": "Buffer",
                },
                "next": null,
              },
              "length": 1,
              "tail": Object {
                "data": Object {
                  "data": Array [
                    49,
                    50,
                    45,
                    70,
                    101,
                    98,
                    45,
                    50,
                    48,
                    49,
                    57,
                    32,
                    48,
                    56,
                    58,
                    53,
                    55,
                    58,
                    52,
                    56,
                    32,
                    101,
                    114,
                    114,
                    111,
                    114,
                    58,
                    32,
                    84,
                    101,
                    115,
                    116,
                    32,
                    101,
                    114,
                    114,
                    111,
                    114,
                    32,
                    109,
                    101,
                    115,
                    115,
                    97,
                    103,
                    101,
                    13,
                    10,
                  ],
                  "type": "Buffer",
                },
                "next": null,
              },
            },
            "decoder": null,
            "defaultEncoding": "utf8",
            "destroyed": false,
            "emitClose": true,
            "emittedReadable": true,
            "encoding": null,
            "endEmitted": false,
            "ended": false,
            "flowing": null,
            "highWaterMark": 16384,
            "length": 48,
            "needReadable": false,
            "objectMode": false,
            "pipes": null,
            "pipesCount": 0,
            "readableListening": false,
            "reading": false,
            "readingMore": true,
            "resumeScheduled": false,
            "sync": false,
          },
          "_transformState": Object {
            "afterTransform": [Function],
            "needTransform": true,
            "transforming": false,
            "writecb": null,
            "writechunk": null,
            "writeencoding": "buffer",
          },
          "_writableState": WritableState {
            "bufferProcessing": false,
            "bufferedRequest": null,
            "bufferedRequestCount": 0,
            "corked": 0,
            "corkedRequestsFree": CorkedRequest {
              "entry": null,
              "finish": [Function],
              "next": null,
            },
            "decodeStrings": true,
            "defaultEncoding": "utf8",
            "destroyed": false,
            "emitClose": true,
            "ended": false,
            "ending": false,
            "errorEmitted": false,
            "finalCalled": false,
            "finished": false,
            "highWaterMark": 16384,
            "lastBufferedRequest": null,
            "length": 0,
            "needDrain": false,
            "objectMode": false,
            "onwrite": [Function],
            "pendingcb": 1,
            "prefinished": false,
            "sync": false,
            "writecb": null,
            "writelen": 0,
            "writing": false,
          },
          "allowHalfOpen": true,
          "domain": null,
          "readable": true,
          "writable": true,
        },
        "_writableState": WritableState {
          "bufferProcessing": false,
          "bufferedRequest": null,
          "bufferedRequestCount": 0,
          "corked": 0,
          "corkedRequestsFree": CorkedRequest {
            "entry": null,
            "finish": [Function],
            "next": null,
          },
          "decodeStrings": true,
          "defaultEncoding": "utf8",
          "destroyed": false,
          "ended": false,
          "ending": false,
          "errorEmitted": false,
          "finalCalled": false,
          "finished": false,
          "highWaterMark": 16,
          "lastBufferedRequest": null,
          "length": 0,
          "needDrain": false,
          "objectMode": true,
          "onwrite": [Function],
          "pendingcb": 2,
          "prefinished": false,
          "sync": false,
          "writecb": null,
          "writelen": 0,
          "writing": false,
        },
        "dirname": ".",
        "domain": null,
        "eol": "
",
        "filename": "errors.log",
        "format": Format {
          "Format": [Function],
          "options": Object {},
        },
        "handleExceptions": undefined,
        "level": "error",
        "levels": Object {
          "debug": 5,
          "error": 0,
          "http": 3,
          "info": 2,
          "silly": 6,
          "verbose": 4,
          "warn": 1,
        },
        "maxFiles": null,
        "maxsize": null,
        "name": "file",
        "options": Object {
          "flags": "a",
        },
        "parent": [Circular],
        "rotationFormat": false,
        "silent": undefined,
        "tailable": false,
        "writable": true,
        "zippedArchive": false,
      },
    ],
    "pipesCount": 3,
    "readableListening": false,
    "reading": false,
    "readingMore": true,
    "resumeScheduled": true,
    "sync": false,
  },
  "_transformState": Object {
    "afterTransform": [Function],
    "needTransform": true,
    "transforming": false,
    "writecb": null,
    "writechunk": null,
    "writeencoding": "utf8",
  },
  "_writableState": WritableState {
    "bufferProcessing": false,
    "bufferedRequest": null,
    "bufferedRequestCount": 0,
    "corked": 0,
    "corkedRequestsFree": CorkedRequest {
      "entry": null,
      "finish": [Function],
      "next": null,
    },
    "decodeStrings": true,
    "defaultEncoding": "utf8",
    "destroyed": false,
    "emitClose": true,
    "ended": false,
    "ending": false,
    "errorEmitted": false,
    "finalCalled": false,
    "finished": false,
    "highWaterMark": 16,
    "lastBufferedRequest": null,
    "length": 0,
    "needDrain": false,
    "objectMode": true,
    "onwrite": [Function],
    "pendingcb": 2,
    "prefinished": false,
    "sync": false,
    "writecb": null,
    "writelen": 0,
    "writing": false,
  },
  "allowHalfOpen": true,
  "defaultMeta": null,
  "domain": null,
  "exceptions": ExceptionHandler {
    "handlers": Map {},
    "logger": [Circular],
  },
  "exitOnError": true,
  "format": Format {
    "options": Object {},
  },
  "level": "info",
  "levels": Object {
    "debug": 5,
    "error": 0,
    "http": 3,
    "info": 2,
    "silly": 6,
    "verbose": 4,
    "warn": 1,
  },
  "profilers": Object {},
  "readable": true,
  "rejections": RejectionHandler {
    "handlers": Map {},
    "logger": [Circular],
  },
  "silent": undefined,
  "writable": true,
}
`;

exports[`Logger test Should call the info method 1`] = `
DerivedLogger {
  "_events": Object {
    "data": Array [
      [Function],
      [Function],
      [Function],
    ],
    "end": Array [
      [Function],
      [Function],
      [Function],
    ],
    "prefinish": [Function],
  },
  "_eventsCount": 3,
  "_maxListeners": undefined,
  "_readableState": ReadableState {
    "awaitDrain": 0,
    "buffer": BufferList {
      "head": null,
      "length": 0,
      "tail": null,
    },
    "decoder": null,
    "defaultEncoding": "utf8",
    "destroyed": false,
    "emitClose": true,
    "emittedReadable": false,
    "encoding": null,
    "endEmitted": false,
    "ended": false,
    "flowing": true,
    "highWaterMark": 16,
    "length": 0,
    "needReadable": true,
    "objectMode": true,
    "pipes": Array [
      Console {
        "__winstonerror": [Function],
        "__winstonwarn": [Function],
        "_events": Object {
          "close": [Function],
          "drain": [Function],
          "error": Array [
            [Function],
            [Function],
          ],
          "finish": [Function],
          "unpipe": Array [
            [Function],
            [Function],
          ],
          "warn": [Function],
        },
        "_eventsCount": 6,
        "_maxListeners": 30,
        "_writableState": WritableState {
          "bufferProcessing": false,
          "bufferedRequest": null,
          "bufferedRequestCount": 0,
          "corked": 0,
          "corkedRequestsFree": CorkedRequest {
            "entry": null,
            "finish": [Function],
            "next": null,
          },
          "decodeStrings": true,
          "defaultEncoding": "utf8",
          "destroyed": false,
          "ended": false,
          "ending": false,
          "errorEmitted": false,
          "finalCalled": false,
          "finished": false,
          "highWaterMark": 16,
          "lastBufferedRequest": null,
          "length": 0,
          "needDrain": false,
          "objectMode": true,
          "onwrite": [Function],
          "pendingcb": 1,
          "prefinished": false,
          "sync": false,
          "writecb": null,
          "writelen": 0,
          "writing": false,
        },
        "consoleWarnLevels": Object {},
        "domain": null,
        "eol": "
",
        "format": Format {
          "Format": [Function],
          "options": Object {},
        },
        "handleExceptions": undefined,
        "level": undefined,
        "levels": Object {
          "debug": 5,
          "error": 0,
          "http": 3,
          "info": 2,
          "silly": 6,
          "verbose": 4,
          "warn": 1,
        },
        "name": "console",
        "parent": [Circular],
        "silent": undefined,
        "stderrLevels": Object {},
        "writable": true,
      },
      File {
        "__winstonerror": [Function],
        "__winstonwarn": [Function],
        "_basename": "info.log",
        "_created": 0,
        "_drain": false,
        "_ending": false,
        "_events": Object {
          "close": [Function],
          "drain": [Function],
          "error": Array [
            [Function],
            [Function],
          ],
          "finish": [Function],
          "unpipe": Array [
            [Function],
            [Function],
          ],
          "warn": [Function],
        },
        "_eventsCount": 6,
        "_maxListeners": undefined,
        "_onError": [Function],
        "_opening": true,
        "_pendingSize": 46,
        "_size": 0,
        "_stream": PassThrough {
          "_events": Object {
            "prefinish": [Function],
          },
          "_eventsCount": 1,
          "_maxListeners": 30,
          "_readableState": ReadableState {
            "awaitDrain": 0,
            "buffer": BufferList {
              "head": Object {
                "data": Object {
                  "data": Array [
                    49,
                    50,
                    45,
                    70,
                    101,
                    98,
                    45,
                    50,
                    48,
                    49,
                    57,
                    32,
                    48,
                    56,
                    58,
                    53,
                    55,
                    58,
                    52,
                    56,
                    32,
                    105,
                    110,
                    102,
                    111,
                    58,
                    32,
                    84,
                    101,
                    115,
                    116,
                    32,
                    105,
                    110,
                    102,
                    111,
                    32,
                    109,
                    101,
                    115,
                    115,
                    97,
                    103,
                    101,
                    13,
                    10,
                  ],
                  "type": "Buffer",
                },
                "next": null,
              },
              "length": 1,
              "tail": Object {
                "data": Object {
                  "data": Array [
                    49,
                    50,
                    45,
                    70,
                    101,
                    98,
                    45,
                    50,
                    48,
                    49,
                    57,
                    32,
                    48,
                    56,
                    58,
                    53,
                    55,
                    58,
                    52,
                    56,
                    32,
                    105,
                    110,
                    102,
                    111,
                    58,
                    32,
                    84,
                    101,
                    115,
                    116,
                    32,
                    105,
                    110,
                    102,
                    111,
                    32,
                    109,
                    101,
                    115,
                    115,
                    97,
                    103,
                    101,
                    13,
                    10,
                  ],
                  "type": "Buffer",
                },
                "next": null,
              },
            },
            "decoder": null,
            "defaultEncoding": "utf8",
            "destroyed": false,
            "emitClose": true,
            "emittedReadable": true,
            "encoding": null,
            "endEmitted": false,
            "ended": false,
            "flowing": null,
            "highWaterMark": 16384,
            "length": 46,
            "needReadable": false,
            "objectMode": false,
            "pipes": null,
            "pipesCount": 0,
            "readableListening": false,
            "reading": false,
            "readingMore": true,
            "resumeScheduled": false,
            "sync": false,
          },
          "_transformState": Object {
            "afterTransform": [Function],
            "needTransform": true,
            "transforming": false,
            "writecb": null,
            "writechunk": null,
            "writeencoding": "buffer",
          },
          "_writableState": WritableState {
            "bufferProcessing": false,
            "bufferedRequest": null,
            "bufferedRequestCount": 0,
            "corked": 0,
            "corkedRequestsFree": CorkedRequest {
              "entry": null,
              "finish": [Function],
              "next": null,
            },
            "decodeStrings": true,
            "defaultEncoding": "utf8",
            "destroyed": false,
            "emitClose": true,
            "ended": false,
            "ending": false,
            "errorEmitted": false,
            "finalCalled": false,
            "finished": false,
            "highWaterMark": 16384,
            "lastBufferedRequest": null,
            "length": 0,
            "needDrain": false,
            "objectMode": false,
            "onwrite": [Function],
            "pendingcb": 1,
            "prefinished": false,
            "sync": false,
            "writecb": null,
            "writelen": 0,
            "writing": false,
          },
          "allowHalfOpen": true,
          "domain": null,
          "readable": true,
          "writable": true,
        },
        "_writableState": WritableState {
          "bufferProcessing": false,
          "bufferedRequest": null,
          "bufferedRequestCount": 0,
          "corked": 0,
          "corkedRequestsFree": CorkedRequest {
            "entry": null,
            "finish": [Function],
            "next": null,
          },
          "decodeStrings": true,
          "defaultEncoding": "utf8",
          "destroyed": false,
          "ended": false,
          "ending": false,
          "errorEmitted": false,
          "finalCalled": false,
          "finished": false,
          "highWaterMark": 16,
          "lastBufferedRequest": null,
          "length": 0,
          "needDrain": false,
          "objectMode": true,
          "onwrite": [Function],
          "pendingcb": 1,
          "prefinished": false,
          "sync": false,
          "writecb": null,
          "writelen": 0,
          "writing": false,
        },
        "dirname": ".",
        "domain": null,
        "eol": "
",
        "filename": "info.log",
        "format": Format {
          "Format": [Function],
          "options": Object {},
        },
        "handleExceptions": undefined,
        "level": "info",
        "levels": Object {
          "debug": 5,
          "error": 0,
          "http": 3,
          "info": 2,
          "silly": 6,
          "verbose": 4,
          "warn": 1,
        },
        "maxFiles": null,
        "maxsize": null,
        "name": "file",
        "options": Object {
          "flags": "a",
        },
        "parent": [Circular],
        "rotationFormat": false,
        "silent": undefined,
        "tailable": false,
        "writable": true,
        "zippedArchive": false,
      },
      File {
        "__winstonerror": [Function],
        "__winstonwarn": [Function],
        "_basename": "errors.log",
        "_created": 0,
        "_drain": false,
        "_ending": false,
        "_events": Object {
          "close": [Function],
          "drain": [Function],
          "error": Array [
            [Function],
            [Function],
          ],
          "finish": [Function],
          "unpipe": Array [
            [Function],
            [Function],
          ],
          "warn": [Function],
        },
        "_eventsCount": 6,
        "_maxListeners": undefined,
        "_onError": [Function],
        "_opening": true,
        "_pendingSize": 0,
        "_size": 0,
        "_stream": PassThrough {
          "_events": Object {
            "prefinish": [Function],
          },
          "_eventsCount": 1,
          "_maxListeners": 30,
          "_readableState": ReadableState {
            "awaitDrain": 0,
            "buffer": BufferList {
              "head": null,
              "length": 0,
              "tail": null,
            },
            "decoder": null,
            "defaultEncoding": "utf8",
            "destroyed": false,
            "emitClose": true,
            "emittedReadable": false,
            "encoding": null,
            "endEmitted": false,
            "ended": false,
            "flowing": null,
            "highWaterMark": 16384,
            "length": 0,
            "needReadable": true,
            "objectMode": false,
            "pipes": null,
            "pipesCount": 0,
            "readableListening": false,
            "reading": false,
            "readingMore": false,
            "resumeScheduled": false,
            "sync": false,
          },
          "_transformState": Object {
            "afterTransform": [Function],
            "needTransform": false,
            "transforming": false,
            "writecb": null,
            "writechunk": null,
            "writeencoding": null,
          },
          "_writableState": WritableState {
            "bufferProcessing": false,
            "bufferedRequest": null,
            "bufferedRequestCount": 0,
            "corked": 0,
            "corkedRequestsFree": CorkedRequest {
              "entry": null,
              "finish": [Function],
              "next": null,
            },
            "decodeStrings": true,
            "defaultEncoding": "utf8",
            "destroyed": false,
            "emitClose": true,
            "ended": false,
            "ending": false,
            "errorEmitted": false,
            "finalCalled": false,
            "finished": false,
            "highWaterMark": 16384,
            "lastBufferedRequest": null,
            "length": 0,
            "needDrain": false,
            "objectMode": false,
            "onwrite": [Function],
            "pendingcb": 0,
            "prefinished": false,
            "sync": true,
            "writecb": null,
            "writelen": 0,
            "writing": false,
          },
          "allowHalfOpen": true,
          "domain": null,
          "readable": true,
          "writable": true,
        },
        "_writableState": WritableState {
          "bufferProcessing": false,
          "bufferedRequest": null,
          "bufferedRequestCount": 0,
          "corked": 0,
          "corkedRequestsFree": CorkedRequest {
            "entry": null,
            "finish": [Function],
            "next": null,
          },
          "decodeStrings": true,
          "defaultEncoding": "utf8",
          "destroyed": false,
          "ended": false,
          "ending": false,
          "errorEmitted": false,
          "finalCalled": false,
          "finished": false,
          "highWaterMark": 16,
          "lastBufferedRequest": null,
          "length": 0,
          "needDrain": false,
          "objectMode": true,
          "onwrite": [Function],
          "pendingcb": 1,
          "prefinished": false,
          "sync": false,
          "writecb": null,
          "writelen": 0,
          "writing": false,
        },
        "dirname": ".",
        "domain": null,
        "eol": "
",
        "filename": "errors.log",
        "format": Format {
          "Format": [Function],
          "options": Object {},
        },
        "handleExceptions": undefined,
        "level": "error",
        "levels": Object {
          "debug": 5,
          "error": 0,
          "http": 3,
          "info": 2,
          "silly": 6,
          "verbose": 4,
          "warn": 1,
        },
        "maxFiles": null,
        "maxsize": null,
        "name": "file",
        "options": Object {
          "flags": "a",
        },
        "parent": [Circular],
        "rotationFormat": false,
        "silent": undefined,
        "tailable": false,
        "writable": true,
        "zippedArchive": false,
      },
    ],
    "pipesCount": 3,
    "readableListening": false,
    "reading": false,
    "readingMore": true,
    "resumeScheduled": true,
    "sync": false,
  },
  "_transformState": Object {
    "afterTransform": [Function],
    "needTransform": true,
    "transforming": false,
    "writecb": null,
    "writechunk": null,
    "writeencoding": "utf8",
  },
  "_writableState": WritableState {
    "bufferProcessing": false,
    "bufferedRequest": null,
    "bufferedRequestCount": 0,
    "corked": 0,
    "corkedRequestsFree": CorkedRequest {
      "entry": null,
      "finish": [Function],
      "next": null,
    },
    "decodeStrings": true,
    "defaultEncoding": "utf8",
    "destroyed": false,
    "emitClose": true,
    "ended": false,
    "ending": false,
    "errorEmitted": false,
    "finalCalled": false,
    "finished": false,
    "highWaterMark": 16,
    "lastBufferedRequest": null,
    "length": 0,
    "needDrain": false,
    "objectMode": true,
    "onwrite": [Function],
    "pendingcb": 1,
    "prefinished": false,
    "sync": false,
    "writecb": null,
    "writelen": 0,
    "writing": false,
  },
  "allowHalfOpen": true,
  "defaultMeta": null,
  "domain": null,
  "exceptions": ExceptionHandler {
    "handlers": Map {},
    "logger": [Circular],
  },
  "exitOnError": true,
  "format": Format {
    "options": Object {},
  },
  "level": "info",
  "levels": Object {
    "debug": 5,
    "error": 0,
    "http": 3,
    "info": 2,
    "silly": 6,
    "verbose": 4,
    "warn": 1,
  },
  "profilers": Object {},
  "readable": true,
  "rejections": RejectionHandler {
    "handlers": Map {},
    "logger": [Circular],
  },
  "silent": undefined,
  "writable": true,
}
`;
